dialect/sql/sqlgraph/errors.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019-present Facebook Inc. All rights reserved.
+// This source code is licensed under the Apache 2.0 license found
+// in the LICENSE file in the root directory of this source tree.
+
+package sqlgraph
+
+import (
+	"strings"
+)
+
+// IsConstraintError returns true if the error resulted from a DB constraint error
+func IsConstraintError(err error) bool {
+	return IsUniquenessConstraintError(err) || IsForeignKeyConstraintError(err)
+}
```

Comment:
```suggestion
// IsConstraintError returns true if the error resulted from a DB constraint error.
func IsConstraintError(err error) bool {
	var e *ConstraintError
	return errors.As(err, &e) || IsUniquenessConstraintError(err) || IsForeignKeyConstraintError(err)
}
```

It does make sense to return true is the error is type `ConstraintError` as well. Also, please add more details regarding the possible constraint errors.